lib/upi-consumer.coffee

```diff
@@ -263,7 +279,7 @@ class UPIConsumer
         else
           throw new Error('Got neither type nor info')
       text:
-        text: "#{if type?.text?.text then ':: '+type.text.text+'\n' else ''}#{info?.text?.text ? ''}"
+        text: (Object.keys(doc).join(", ") + "\nBLERG 0: " + doc[Object.keys(doc)[0]] + "\nBLERG 1: " + doc[Object.keys(doc)[1]] + "\nBLERG 1 keys: " + Object.keys(doc[Object.keys(doc)[1]]) + "\nBLERG 1[0]: " + doc[Object.keys(doc)[1]]["text"] + "--(Karl was here)\n") + "#{if type?.text?.text then ':: '+type.text.text+'\n' else ''}#{info?.text?.text ? ''}"
```

Comment:
Hi, I have created minimal test cases that demonstrate various failure modes of "Jump to Declaration" for both `haskell-ghc-mod` and `ide-haskell-hasktags` in [this demo project](https://github.com/kostmo/atom-ghc-mod-testcases/blob/master/stack-multi-package/my-package-b/src/LibB1.hs#L15-L45).

In summary:

* Indeed, `haskell-ghc-mod` never works "across" packages
* `ide-haskell-hasktags` works both within the same package and "across" packages when:
    * the symbol is *not* highlighted; one must place the cursor within the symbol without highlighting it
    * the symbol is *not qualified* by a namespace; it must be "bare"
* `haskell-ghc-mod` does not work when a symbol is qualified by a *renamed* import, unless one *manually highlights* the symbol alone while excluding the leading namespace and period.

Are the latter two items known limitations?